actionpack/test/controller/request_forgery_protection_test.rb

```diff
@@ -840,6 +840,19 @@ def test_rejects_token_for_incorrect_method_button_to
     end
   end
 
+  def test_rejects_global_csrf_token
+    get :index
+
+    token = @controller.send(:form_authenticity_token)
+
+    # This is required because PATH_INFO isn't reset between requests.
+    @request.env["PATH_INFO"] = "/per_form_tokens/post_one"
+    assert_raises(ActionController::InvalidAuthenticityToken) do
```

Comment:
This whole test was moved above, but this is the only line which was actually changed.